HPXMLtoOpenStudio/resources/hpxml_schematron/EPvalidator.xml

```diff
@@ -95,6 +96,79 @@
+      <sch:assert role='ERROR' test='count(h:UtilityRate[h:FuelType="natural gas"]/h:FixedCharge) &lt;= 1'>Expected 0 or 1 element(s) for xpath: UtilityRate[FuelType="natural gas"]/FixedCharge</sch:assert>
+      <sch:assert role='ERROR' test='count(h:UtilityRate[h:FuelType="natural gas"]/h:MarginalRate) &lt;= 1'>Expected 0 or 1 element(s) for xpath: UtilityRate[FuelType="natural gas"]/MarginalRate</sch:assert>
+    </sch:rule>
+  </sch:pattern>
+
+  <sch:pattern>
+    <sch:title>[UtilityRate=PropaneAndFuelOil]</sch:title>
+    <sch:rule context='/h:HPXML/h:SoftwareInfo/h:extension/h:UtilityBillScenarios/h:UtilityBillScenario/h:UtilityRate[h:FuelType[text()="propane" or h:FuelType="fuel oil"]]'>
+    <sch:assert role='ERROR' test='count(h:UtilityRate[h:FuelType="propane"]/h:MarginalRate) &lt;= 1'>Expected 0 or 1 element(s) for xpath: UtilityRate[FuelType="propane"]/MarginalRate</sch:assert>
+    <sch:assert role='ERROR' test='count(h:UtilityRate[h:FuelType="fuel oil"]/h:MarginalRate) &lt;= 1'>Expected 0 or 1 element(s) for xpath: UtilityRate[FuelType="fuel oil"]/MarginalRate</sch:assert>
+    </sch:rule>
+  </sch:pattern>
+
+  <sch:pattern>
+    <sch:title>[UtilityRate=OtherFuel]</sch:title>
+    <sch:rule context='/h:HPXML/h:SoftwareInfo/h:extension/h:UtilityBillScenarios/h:UtilityBillScenario/h:UtilityRate[h:FuelType[text()="coal" or h:FuelType="wood" or h:FuelType="wood pellets"]]'>
+      <sch:assert role='ERROR' test='count(h:MarginalRate) = 1'>Expected 1 element(s) for xpath: MarginalRate</sch:assert>
+    </sch:rule>
+  </sch:pattern>
+
+  <sch:pattern>
+    <sch:title>[PVCompensation]</sch:title>
+    <sch:rule context='/h:HPXML/h:SoftwareInfo/h:extension/h:UtilityBillScenarios/h:UtilityBillScenario/h:PVCompensation'>
+      <sch:assert role='ERROR' test='count(h:CompensationType) = 1'>Expected 1 element(s) for xpath: CompensationType</sch:assert> <!-- See [UtilityRate=NetMetering] or [UtilityRate=FeedInTariff] -->
+      <sch:assert role='ERROR' test='h:CompensationType[text()="Net Metering" or text()="Feed-In Tariff"] or not(h:CompensationType)'>Expected CompensationType to be 'Net Metering' or 'Feed-In Tariff'</sch:assert>
+      <sch:assert role='ERROR' test='count(h:MonthlyGridConnectionFeeUnits) &lt;= 1'>Expected 0 or 1 element(s) for xpath: MonthlyGridConnectionFeeUnits</sch:assert>
+      <sch:assert role='ERROR' test='h:MonthlyGridConnectionFeeUnits[text()="$/kW" or text()="$"] or not(h:CompensationType)'>Expected MonthlyGridConnectionFeeUnits to be '$/kW' or '$'</sch:assert>
+      <sch:assert role='ERROR' test='count(h:MonthlyGridConnectionFeeUnits) + count(h:MonthlyGridConnectionFee) = 0 or count(h:MonthlyGridConnectionFeeUnits) + count(h:MonthlyGridConnectionFee) = 2'>Expected 0 or 1 element(s) for xpath: MonthlyGridConnectionFeeUnits | MonthlyGridConnectionFee</sch:assert>
```

Comment:
Should say "Expected 0 or 2 element(s)"...

But even better would be to change this to match how Units/Values are typically handled in HPXML:
```
  <MonthlyGridConnectionFee>
    <Units>XXX</Units>
    <Value>XXX</Value>
  </MonthlyGridConnectionFee>
```

See how OS-HPXML handles these kinds of inputs by looking at heating/cooling system efficiencies, for example.

Reply:
I'm of two minds here. It's probably helpful as it makes the units clearer, but I can't imagine we would ever offer additional units and it makes the section in the HPXML file much longer. And we'd have to it for `NetMeteringAnnualExcessSellbackRate` too.

I guess I lean towards not making that change so we can wrap this up and move on.